Deleting event from search index fails if the event is not in the search index

Deleting the event from the index fails with OpenSearch returing that the event is already gone which causes Opencast to completely fail the operation. It should be happy that it's already gone instead and continue.

```
2024-09-18T18:04:27,215 | INFO  | (SearchServiceIndex:351) - Marking media package 0924ee2d-3530-4f3a-9f81-296c29008579 as deleted in search index                                                                                                                                                                                                                                                                                                 18:04:27 [243/271]
2024-09-18T18:04:27,248 | INFO  | (ServiceRegistryJpaImpl:2268) - State set to WARNING for current service org.opencastproject.search on host https://oc-admin.virtuos.uni-osnabrueck.de                                                                                                                                                                                                 
2024-09-18T18:04:27,284 | ERROR | (AbstractJobProducer$JobRunner:344) - Error handling operation 'Delete':                                                                                                                                                                                                                                                                               
org.opencastproject.serviceregistry.api.ServiceRegistryException: Error handling operation 'Delete'                                                                                                                                                                                                                                                                                      
        at org.opencastproject.search.impl.SearchServiceImpl.process(SearchServiceImpl.java:291) ~[?:?]                                                                                                                                                                                                                                                                                  
        at org.opencastproject.job.api.AbstractJobProducer$JobRunner.call(AbstractJobProducer.java:314) [!/:?]                                                                                                                                                                                                                                                                           
        at org.opencastproject.job.api.AbstractJobProducer$JobRunner.call(AbstractJobProducer.java:273) [!/:?]                                                                                                                                                                                                                                                                           
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]                                                                                                                                                                                                                                                                                                                
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]                                                                                                                                                                                                                                                                                         
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]                                                                                                                                                                                                                                                                                         
        at java.lang.Thread.run(Thread.java:829) [?:?]                                                                                                                                                                                                                                                                                                                                   
Caused by: org.elasticsearch.ElasticsearchStatusException: Elasticsearch exception [type=document_missing_exception, reason=[_doc][0924ee2d-3530-4f3a-9f81-296c29008579]: document missing]                                                                                                                                                                                              
        at org.elasticsearch.rest.BytesRestResponse.errorFromXContent(BytesRestResponse.java:187) ~[?:?]                                                                                                                                                                                                                                                                                 
        at org.elasticsearch.client.RestHighLevelClient.parseEntity(RestHighLevelClient.java:1911) ~[?:?]                                                                                                                                                                                                                                                                                
        at org.elasticsearch.client.RestHighLevelClient.parseResponseException(RestHighLevelClient.java:1888) ~[?:?]                                                                                                                                                                                                                                                                     
        at org.elasticsearch.client.RestHighLevelClient.internalPerformRequest(RestHighLevelClient.java:1645) ~[?:?]                                                                                                                                                                                                                                                                     
        at org.elasticsearch.client.RestHighLevelClient.performRequest(RestHighLevelClient.java:1602) ~[?:?]                                                                                                                                                                                                                                                                             
        at org.elasticsearch.client.RestHighLevelClient.performRequestAndParseEntity(RestHighLevelClient.java:1572) ~[?:?]                                                                                                                                                                                                                                                               
        at org.elasticsearch.client.RestHighLevelClient.update(RestHighLevelClient.java:1039) ~[?:?]                                                                                                                                                                                                                                                                                     
        at org.opencastproject.search.impl.SearchServiceIndex.deleteSynchronously(SearchServiceIndex.java:357) ~[?:?]                                                                                                                                                                                                                                                                    
        at org.opencastproject.search.impl.SearchServiceImpl.lambda$process$1(SearchServiceImpl.java:274) ~[?:?]                                                                                                                                                                                                                                                                         
        at org.opencastproject.security.util.SecurityUtil.runAs(SecurityUtil.java:79) ~[?:?]                                                                                                                                                                                                                                                                                             
        at org.opencastproject.search.impl.SearchServiceImpl.process(SearchServiceImpl.java:273) ~[?:?]                                                                                                                                                                                                                                                                                  
        ... 6 more                                                                                                                                                                                                                                                                                                                                                                       
        Suppressed: org.elasticsearch.client.ResponseException: method [POST], host [https://oc-opensearch.uni-osnabrueck.de:443], URI [/opencast_search/_update/0924ee2d-3530-4f3a-9f81-296c29008579?timeout=1m], status line [HTTP/1.1 404 Not Found]                                                                                                                                  
{"error":{"root_cause":[{"type":"document_missing_exception","reason":"[_doc][0924ee2d-3530-4f3a-9f81-296c29008579]: document missing","index":"opencast_search","shard":"0","index_uuid":"gOLGKcUIQq6ssqc9celKYw"}],"type":"document_missing_exception","reason":"[_doc][0924ee2d-3530-4f3a-9f81-296c29008579]: document missing","index":"opencast_search","shard":"0","index_uuid":"gOLGKcUIQq6ssqc9celKYw"},"status":404}
                at org.elasticsearch.client.RestClient.convertResponse(RestClient.java:326) ~[?:?]                                                                                                                                                                                                                                                                                       
                at org.elasticsearch.client.RestClient.performRequest(RestClient.java:296) ~[?:?]                                                                                                                                                                                                                                                                                        
                at org.elasticsearch.client.RestClient.performRequest(RestClient.java:270) ~[?:?]                                                                                                                                                                                                                                                                                        
                at org.elasticsearch.client.RestHighLevelClient.internalPerformRequest(RestHighLevelClient.java:1632) ~[?:?]                                                                                                                                                                                                                                                             
                at org.elasticsearch.client.RestHighLevelClient.performRequest(RestHighLevelClient.java:1602) ~[?:?]                                                                                                                                                                                                                                                                     
                at org.elasticsearch.client.RestHighLevelClient.performRequestAndParseEntity(RestHighLevelClient.java:1572) ~[?:?]                                                                                                                                                                                                                                                       
                at org.elasticsearch.client.RestHighLevelClient.update(RestHighLevelClient.java:1039) ~[?:?]                                                                                                                                                                                                                                                                             
                at org.opencastproject.search.impl.SearchServiceIndex.deleteSynchronously(SearchServiceIndex.java:357) ~[?:?]                                                                                                                                                                                                                                                            
                at org.opencastproject.search.impl.SearchServiceImpl.lambda$process$1(SearchServiceImpl.java:274) ~[?:?]                                                                                                                                                                                                                                                                 
                at org.opencastproject.security.util.SecurityUtil.runAs(SecurityUtil.java:79) ~[!/:?]                                                                                                                                                                                                                                                                                    
                at org.opencastproject.search.impl.SearchServiceImpl.process(SearchServiceImpl.java:273) ~[?:?]                                                                                                                                                                                                                                                                          
                at org.opencastproject.job.api.AbstractJobProducer$JobRunner.call(AbstractJobProducer.java:314) [!/:?]                                                                                                                                                                                                                                                                   
                at org.opencastproject.job.api.AbstractJobProducer$JobRunner.call(AbstractJobProducer.java:273) [!/:?]                                                                                                                                                                                                                                                                   
                at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]                                                                                                                                                                                                                                                                                                        
                at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]                                                                                                                                                                                                                                                                                 
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]                                                                                                                                                                                                                                                                                 
                at java.lang.Thread.run(Thread.java:829) [?:?]                                                                                                                                                                                                                                                                                                                           
2024-09-18T18:04:31,110 | ERROR | (WorkflowOperationWorker:140) - Workflow operation 'operation:'retract-engage, state:'FAILED'' failed                                                                                                                                                                                                                                                  
org.opencastproject.workflow.api.WorkflowOperationException: org.opencastproject.workflow.api.WorkflowOperationException: Removing media package from search did not complete successfully                                                                                                                                                                                               
        at org.opencastproject.workflow.handler.distribution.RetractEngageWorkflowOperationHandler.start(RetractEngageWorkflowOperationHandler.java:209) ~[?:?]                                                                                                                                                                                                                          
        at org.opencastproject.workflow.impl.WorkflowOperationWorker.start(WorkflowOperationWorker.java:212) ~[!/:?]                                                                                                                                                                                                                                                                     
        at org.opencastproject.workflow.impl.WorkflowOperationWorker.execute(WorkflowOperationWorker.java:117) [!/:?]                                                                                                                                                                                                                                                                    
        at org.opencastproject.workflow.impl.WorkflowServiceImpl.runWorkflowOperation(WorkflowServiceImpl.java:722) [!/:?]                                                                  
        at org.opencastproject.workflow.impl.WorkflowServiceImpl.process(WorkflowServiceImpl.java:1755) [!/:?]                                                                              
        at org.opencastproject.workflow.impl.WorkflowServiceImpl$JobRunner.call(WorkflowServiceImpl.java:2116) [!/:?]                                                                       
        at org.opencastproject.workflow.impl.WorkflowServiceImpl$JobRunner.call(WorkflowServiceImpl.java:2082) [!/:?]                                                                       
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]                                                                                                                   
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]                                                                                                                                  
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]                                                                                            
        at java.lang.Thread.run(Thread.java:829) [?:?]                                                                                                                                      
Caused by: org.opencastproject.workflow.api.WorkflowOperationException: Removing media package from search did not complete successfully                                                                                          
        at org.opencastproject.workflow.handler.distribution.RetractEngageWorkflowOperationHandler.start(RetractEngageWorkflowOperationHandler.java:192) ~[?:?]                             
        ... 10 more 
```

--- a/modules/search-service-impl/src/main/java/org/opencastproject/search/impl/SearchServiceIndex.java
+++ b/modules/search-service-impl/src/main/java/org/opencastproject/search/impl/SearchServiceIndex.java
@@ -355,7 +355,11 @@ public final class SearchServiceIndex extends AbstractIndexProducer implements I
       var updateRequst = new UpdateRequest(INDEX_NAME, mediaPackageId)
           .doc(gson.toJson(json), XContentType.JSON);
       esIndex.getClient().update(updateRequst, RequestOptions.DEFAULT);
-
+    } catch (ElasticsearchStatusException e) {
+      if (e.status().getStatus() != RestStatus.NOT_FOUND.getStatus()) {
+        throw e;
+      }
+      logger.warn("Event {} is not in the search index. Skipping deletion", mediaPackageId);
     } catch (IOException e) {
       throw new SearchException("Could not delete episode " + mediaPackageId + " from index", e);
     }

